commit fd85ac9cc623847e9d94d9f9241ab34e2c146cbf
Author: Luiz Angelo Daros de Luca <luizluca@gmail.com>
Date:   Thu Oct 25 17:39:52 2018 -0300

    * src/host.c (sufmatch): Fix dot-prefixed domain matching
    
    Current sufmatch does not match when domain is dot-prefixed.
    The example of no_proxy in man (.mit.edu) does use a dot-prefixed
    domain.
    
    Signed-off-by: Luiz Angelo Daros de Luca <luizluca@gmail.com>
    Copyright-paperwork-exempt: Yes

diff --git a/src/host.c b/src/host.c
index b42cd6e8..2bf848f3 100644
--- a/src/host.c
+++ b/src/host.c
@@ -1033,8 +1033,9 @@ sufmatch (const char **list, const char *what)
       /* Domain or subdomain match
        * k == -1: exact match
        * k >= 0 && what[k] == '.': subdomain match
+       * k >= 0 && list[i][0] == '.': dot-prefixed subdomain match
        */
-      if (j == -1 && (k == -1 || what[k] == '.'))
+      if (j == -1 && (k == -1 || what[k] == '.' || list[i][0] == '.'))
         return true;
     }
 
